### PR TITLE
(fix): introduce extended properties into the IR

### PR DESCRIPTION
**Pull request by dsinghvi**
This PR will introduce extended properties into the IR to make it easier for generators to inline properties and handle extensions.

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/extended-properties/addExtendedPropertiesToIr.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    IntermediateRepresentation,
+    ObjectProperty,
+    ObjectTypeDeclaration,
+    TypeDeclaration
+} from "@fern-api/ir-sdk";
+import { LoggableFernCliError } from "@fern-api/task-context";
+
+type TypesAndServices = Pick<IntermediateRepresentation, "types" | "services">;
+
+export function addExtendedPrpoertiesToIr(ir: TypesAndServices): TypesAndServices {
```
Comment (dcb6) [inline packages/cli/generation/ir-generator/src/extended-properties/addExtendedPropertiesToIr.ts:13]:
```suggestion
export function addExtendedPropertiesToIr(ir: TypesAndServices): TypesAndServices {
```

Comment (dcb6) [reply packages/cli/generation/ir-generator/src/extended-properties/addExtendedPropertiesToIr.ts:13]: few other of the same lil typo

File: packages/cli/generation/ir-generator/src/extended-properties/addExtendedPropertiesToIr.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    declaredTypeName: DeclaredTypeName,
+    ir: TypesAndServices
+): ObjectProperty[] {
+    const objectTypeDeclaration = getObjectTypeDeclarationFromTypeId(declaredTypeName.typeId, ir);
+    return getAllPropertiesForObject({ objectTypeDeclaration, ir });
+}
+
+function getObjectTypeDeclarationFromTypeId(typeId: string, ir: TypesAndServices): ObjectTypeDeclaration {
+    const maybeType = ir.types[typeId];
+    if (maybeType?.shape.type !== "object") {
```
Comment (dcb6) [inline packages/cli/generation/ir-generator/src/extended-properties/addExtendedPropertiesToIr.ts:88]: this couldn't be alias?

Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/extended-properties/addExtendedPropertiesToIr.ts]: great call

**Pull request merged**